backend/vault.go

```diff
@@ -15,22 +18,46 @@ import (
 
 var vMetrics *vaultMetrics
 
-const defaultSecretKey = "data"
+const (
+	defaultSecretKey       = "data"
+	kubernetesJwtTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	kubernetesAuthMethod   = "kubernetes"
+	appRoleAuthMethod      = "approle"
+)
 
 type client struct {
 	vclient            *api.Client
 	logical            *api.Logical
 	roleID             string
+	authMethod         string
 	secretID           string
+	kubernetesRole     string
 	maxTokenTTL        int64
 	tokenPollingPeriod time.Duration
 	renewTTLIncrement  int
 	engine             engine
 	approlePath        string
+	kubernetesPath     string
 	logger             logr.Logger
 }
 
 func (c *client) vaultLogin() error {
```

Comment:
could you please add new TestVaultLogin* test classes corresponding the new test scenario?

Reply:
not sure what you mean, kubernetes login is coveres with new test cases.

Reply:
I mean covering this switch logic that is responsible to decide which method we should use

Reply:
ok :)

Reply:
Isn't that covered on every call to `vaultClient()` ?

Reply:
yes, partially, just the old version.

The new case is not covered since we are just executing the **vaultLogin** method for the switch's default case.

Reply:
ok, got you. I guess there's a caveat, since to use kubernetes auth we need to read the SA token from disk. I used `io.Reader` passed directly to the `vaultLoginKubernetes()` method. So this is going to be tricky.

Reply:
if this is kind of a block right now, no worries. Feel free to visit it in a different PR.
But it's interesting to have it covered.

If you prefer to do it separately, please, open us an issue to have it in mind.

Reply:
If this is enough to you to approve the PR, we can tackle this in a separate PR. Like you said, I think it's interesting, but not too critical.

Reply:
perfect, let's do it in another iteration.